student/models.py

```diff
@@ -1,3 +1,30 @@
-# from django.db import models
+from django.db import models
+from django.conf import settings
 
-# Create your models here.
+
+class Gender(models.TextChoices):
+    MALE = "M", "Male"
+    FEMALE = "F", "Female"
+    UNSPECIFIED = "U", "Unspecified"
+
+
+class EducationalInstitution(models.Model):
+    institution_name = models.CharField(max_length=255)
+
+    def __str__(self):
+        return self.institution_name
+
+
+class Student(models.Model):
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.CASCADE, primary_key=True)
+    first_name = models.CharField(max_length=255)
+    last_name = models.CharField(max_length=255)
+    email = models.EmailField(max_length=255)
+    date_of_birth = models.DateField()
+    gender = models.CharField(max_length=1, choices=Gender.choices, default=Gender.UNSPECIFIED)
+    phone_number = models.CharField(max_length=10, blank=True, null=True)
+    educational_institution_name = models.ManyToManyField(EducationalInstitution, related_name='students')
```

Comment:
Bad field name - its not the name of the institute - its a link to multiple records (Did you really mean to put an m-to-m relatoion here? Maybe you meant one-to-m?).

Reply:
I wanted a many to many relation so every institution can access the list of students that are enrolled to it, as well as every student can be enrolled to multiple institutions.. From what I understood if I made a one to many relation it wouldn't allow an institution to be related to multiple students? Or am I missing something?

Reply:
Maybe we can change it to "current_institution" or something like that since it is a student jobs site so the relevant information on the student would be where he is studying now? And make it a one to many relation for the institutions table. I feel like where he studied in the past is less relevant and can be in his CV.

Reply:
I tend to agree with Eldar here, it seems to me that if you link to many institutions you need extra data to explain each link (You studied there in the past and graduated, etc.), but linking to one that is the one you're currently actively enrolled to seems to tell you the full story.

One thing you should consider there - is it possible for a user of the system not to be enrolled at all?

Reply:
I don't think it should be possible for a user to not be enrolled at all as the website is intended for students.
I'll create two fields, one for the current place the student is enrolled at and one for any previous places.
So I guess the current place should be a Foreign key field and the previous places should be many to many?

Reply:
I seem to remember you or some other members of your team did not want to go towards a fully structured CV information, so why do you need this for academic institutes?

Everything you put in the model goes towards providing some application feature, what are the features that would be based on the fields that you propose?